### PR TITLE
[Feature Store] Add kafka source to ingestion service

### PR DESCRIPTION
**Pull request by dinal**
pending https://github.com/nuclio/nuclio-jupyter/pull/129 and next nuclio-jupyter release

### CODE REVIEW

File: mlrun/datastore/sources.py [hunk 1]
```diff
@@ -350,12 +350,55 @@ def __init__(
...
+       :parameter sasl_pass: Optional, password to use for sasl authentications
+    """
+
+    kind = "kafka"
+
+    def __init__(
+        self,
+        brokers="localhost:9092",
+        topics="topic",
+        group="serving",
```
Comment (Hedingber) [inline mlrun/datastore/sources.py:371]: Are you sure we want these defaults ? 
seems like a left over from testing or something

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: mlrun/datastore/sources.py (hunk 1 above)
Comment (dinal) [reply mlrun/datastore/sources.py:371]: same as we have in stream 
https://github.com/mlrun/mlrun/blob/1311e51857e0d029bf0c030c06e9b67fec7d771d/mlrun/datastore/sources.py#L337
it must have some defaults

File: mlrun/feature_store/ingestion.py [hunk 2]
```diff
@@ -243,6 +245,23 @@ def add_source_trigger(source, function):
             source.attributes["seek_to"],
             source.attributes["shards"],
         )
+    if isinstance(source, KafkaSource):
```
Comment (yaronha) [inline mlrun/feature_store/ingestion.py:248]: @dinal why not add some method to `KafkaSource` (and v3io), which returns/adds the nuclio trigger, this way the ingestion is clean and we can add more triggers in a simpler way ..

File: mlrun/feature_store/ingestion.py [hunk 3]
```diff
@@ -243,6 +245,23 @@ def add_source_trigger(source, function):
...
+        func = function.add_trigger("kafka", trigger)
+        sasl_user = source.attributes.get("sasl_user")
+        sasl_pass = source.attributes.get("sasl_pass")
+        if sasl_user and sasl_pass:
+            trigger.sasl(sasl_user, sasl_pass)
+        replicas = 1 if not partitions else len(partitions)
+        func.spec.min_replicas = replicas
+        func.spec.max_replicas = replicas
     else:
         raise mlrun.errors.MLRunInvalidArgumentError(
```
Comment (yaronha) [inline mlrun/feature_store/ingestion.py:266]: this `raise` can go to the source base class

**Pull request merged**